src/ethereum/dai.rs

```diff
@@ -132,7 +134,12 @@ impl Amount {
         // Apply the rate
         let (worth, _remainder) = self.as_atto().div_rem(&uint_rate);
 
+        if worth.is_zero() {
```

Comment:
You can add a proptest on this function to ensure there is no other arithmetic issues.

Reply:
This will all get sorted when we change to price quantity